pkg/webhook/membercluster/membercluster_validating_webhook.go

```diff
@@ -22,26 +25,47 @@ var (
 )
 
 type memberClusterValidator struct {
+	client  client.Client
 	decoder webhook.AdmissionDecoder
 }
 
 // Add registers the webhook for K8s bulit-in object types.
 func Add(mgr manager.Manager) error {
 	hookServer := mgr.GetWebhookServer()
-	hookServer.Register(ValidationPath, &webhook.Admission{Handler: &memberClusterValidator{admission.NewDecoder(mgr.GetScheme())}})
+	hookServer.Register(ValidationPath, &webhook.Admission{Handler: &memberClusterValidator{client: mgr.GetClient(), decoder: admission.NewDecoder(mgr.GetScheme())}})
 	return nil
 }
 
 // Handle memberClusterValidator checks to see if member cluster has valid fields.
-func (v *memberClusterValidator) Handle(_ context.Context, req admission.Request) admission.Response {
+func (v *memberClusterValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
Another naming nit: the added logic might no longer belong in a validator? It's more of a guard 😄

Reply:
That is, the logic seems to be more aligned with those in `fleetresourcehandler` pkg?

Reply:
fleetresourcehandler pkg is aim to validate the CR permissions so far. feel this package is better?